### ISSUE TITLE
get_report_document, Unable to start download due to error with zlib.decompress

### ISSUE THREAD

**Issue by BeboGhattas** [saleweaver/python-amazon-sp-api#879]
There is an error with Reports / get_report_document, Unable to proceed to download a report (Code was working until recently)
issue at `document = zlib.decompress(bytearray(document), 15 + 32)`

Error output as below:
```python
error                                     Traceback (most recent call last)
Cell In [1], line 44
     41 output_file = report_document_id + ".json"
     43 # get report document
---> 44 getReportDocumentResponse = Reports(credentials=credentials,
     45     marketplace=Marketplaces.US).get_report_document(
     46         reportDocumentId=report_document_id,
     47         decrypt=True,
     48         file=output_file,
     49         character_code='utf-8'
     50         )

File c:\Users\english\AppData\Local\Programs\Python\Python310\lib\site-packages\sp_api\base\helpers.py:19, in sp_endpoint.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
     14 def wrapper(*args, **kwargs):
...
--> 369     document = zlib.decompress(bytearray(document), 15 + 32)
    370 document = document.decode(character_code)
    371 if download:

error: Error -5 while decompressing data: incomplete or truncated stream
```

Full code if needed to reproduce the error:
```python
credentials=dict(
        refresh_token='Atzr***************',
        lwa_app_id='amzn1.application-***********************',
        lwa_client_secret='***********************************',
        aws_secret_key='*************************',
        aws_access_key='**************************',
        role_arn='arn:aws:iam::********************:role/SellingPartnerAPIRole',
    )

# report request
createReportResponse = Reports(credentials=credentials
    ,marketplace=Marketplaces.US).create_report(
        reportType="GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT",
        reportOptions={"reportPeriod": "WEEk"}, #DAY, WEEK, MONTH, and QUARTER
        dataStartTime="2023-01-01", #YYYY-MM-DD
        dataEndTime="2023-01-07"
        )

# report id
report_id = createReportResponse.payload['reportId']
print("Report ID: ", report_id)

# get report (loop)
getReportResponse = Reports(credentials=credentials, marketplace=Marketplaces.US).get_report(reportId=report_id)
while getReportResponse.payload['processingStatus'] != 'DONE':
    print("Report status is: %s. Sleeping for 10 seconds..." % getReportResponse.payload['processingStatus'])
    time.sleep(30)
    getReportResponse = Reports(credentials=credentials, marketplace=Marketplaces.US).get_report(reportId=report_id)

# report document id
report_document_id = getReportResponse.payload['reportDocumentId']
print("Report document ID: ", report_document_id)

# create filename
output_file = report_document_id + ".json"

# get report document
getReportDocumentResponse = Reports(credentials=credentials,
    marketplace=Marketplaces.US).get_report_document(
        reportDocumentId=report_document_id,
        decrypt=True,
        file=output_file,
        character_code='utf-8'
        )
```

**Comment by saleweaver** [comment]
Just to be specific, does it work if you pass in one of the Enum values instead of "WEEk"?

The report may be quite large, might be a problem with this too. I'll have a look, but please verify with above change.

**Comment by chocobn69** [comment]
Got a problem with this too : I'm passing `download=False` + a file object to not load file in memory, but `decompress` call is loading : Memory error

**Comment by chocobn69** [comment]
NVM I think it's not the same problem here.

**Comment by BeboGhattas** [comment]
> Just to be specific, does it work if you pass in one of the Enum values instead of "WEEk"? (There's a small typo in reportOptions)
> 
> The report may be quite large, might be a problem with this too. I'll have a look, but please verify with above change.

The download file is estimated 4-5 GB, it normally take 40-45 mins to since script run to get the output
However, I just tried to run it on several periods, It worked only when the report is smaller (Today's was only 3GB)

So this may continue to occur if the report is large/larger than it used to be

**Comment by Isaac-Tolu** [comment]
I was trying to get this same report. When I run this, my server kills the process when it reaches the decrypting stage.I checked and I saw that it uses my entire CPU. I was wondering if anyone faced CPU issues when trying to get this report.

**Comment by A-Posthuman** [comment]
Hi, just to chime in I'm also running into the same issue getting these search terms reports, my server kills the process, I think due to running out of mem, but not sure. It happens around the time my program calls get_report_document().

Any workarounds?

**Comment by thinhnd2104** [comment]
In my case, instead of decompressing the report when receiving all the data you can save the compressed data and import directly to the database .

**Comment by PoByBolek** [comment]
I have the same issue since a few days ago. If this helps, here are the request and response headers for the metadata request:
```
GET https://sellingpartnerapi-eu.amazon.com/reports/2021-06-30/documents/amzn1.spdoc.1.4.eu.<redacted> HTTP/1.1
host: sellingpartnerapi-eu.amazon.com
user-agent: python-sp-api-0.17.5
x-amz-access-token: <redacted>
x-amz-date: 20230627T160337Z
content-type: application/json



HTTP/1.1 200 OK
Server: Server
Date: Tue, 27 Jun 2023 16:03:37 GMT
Content-Type: application/json
Content-Length: 539
Connection: keep-alive
x-amz-rid: <redacted>
x-amzn-RateLimit-Limit: 0.0167
x-amzn-RequestId: <redacted>
x-amz-apigw-id: <redacted>
X-Amzn-Trace-Id: Root=<redacted>
Vary: Content-Type,Accept-Encoding,User-Agent
Strict-Transport-Security: max-age=47474747; includeSubDomains; preload

{
    "reportDocumentId": "amzn1.spdoc.1.4.eu.<redacted>",
    "compressionAlgorithm": "GZIP",
    "url": "https://tortuga-prod-eu.s3-eu-west-1.amazonaws.com/%2FNinetyDays/amzn1.tortuga.4.eu.<redacted>?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230627T160337Z&X-Amz-SignedHeaders=host&X-Amz-Expires=300&X-Amz-Credential=<redacted>%2F20230627%2Feu-west-1%2Fs3%2Faws4_request&X-Amz-Signature=<redacted>"
}
```

And the request and respone headers for the actual report document:
```
GET https://tortuga-prod-eu.s3-eu-west-1.amazonaws.com/%2FNinetyDays/amzn1.tortuga.4.eu.<redacted>?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20230627T160337Z&X-Amz-SignedHeaders=host&X-Amz-Expires=300&X-Amz-Credential=<redacted>%2F20230627%2Feu-west-1%2Fs3%2Faws4_request&X-Amz-Signature=<redacted> HTTP/1.1



HTTP/1.1 200 OK
x-amz-id-2: <redacted>
x-amz-request-id: <redacted>
Date: Tue, 27 Jun 2023 16:03:38 GMT
Last-Modified: Thu, 22 Jun 2023 23:30:48 GMT
x-amz-expiration: expiry-date="Thu, 21 Sep 2023 00:00:00 GMT", rule-id="Rule for: NinetyDays"
ETag: "<redacted>"
x-amz-server-side-encryption: aws:kms
x-amz-server-side-encryption-aws-kms-key-id: arn:aws:kms:eu-west-1:<redacted>:key/<redacted>
x-amz-server-side-encryption-bucket-key-enabled: true
Content-Encoding: gzip
x-amz-version-id: <redacted>
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Server: AmazonS3
Content-Length: 132228

{"reportSpecification":{"reportType":"GET_SALES_AND_TRAFFIC_REPORT","reportOptions":{"dateGranularity":"DAY","asinGranularity":"CHILD"},"dataStartTime":"2023-06-21","dataEndTime":"2023-06-21","marketplaceIds":["A1PA6795UKMFR9"]},"salesAndTrafficByDate":[...
```

Looks like everyone thinks that the document should be GZip compressed but it actually isn't...

**Comment by PoByBolek** [comment]
> Looks like the Amazon servers think that the document should be GZip compressed but it actually isn't...

Or rather `requests` (or the underlying `urllib3`) is already decompressing the gzipped content for us because of the `Content-Encoding` header?

https://github.com/psf/requests/blob/22db55a8896b69e53d0a3cc2764c27b832c81478/requests/models.py#L812-L820

```python
        def generate():
            # Special case for urllib3.
            if hasattr(self.raw, "stream"):
                try:
                    yield from self.raw.stream(chunk_size, decode_content=True)
                except ProtocolError as e:
                    raise ChunkedEncodingError(e)
                except DecodeError as e:
                    raise ContentDecodingError(e)
```

**Comment by PoByBolek** [comment]
This seems to have been fixed with #1079 (commit d15cb24c692ba18932450b01e62abad4721cfe64) in [`v0.18.5`](https://github.com/saleweaver/python-amazon-sp-api/releases/tag/v0.18.5), right?
